__init__.py

```diff
@@ -607,17 +627,44 @@ def __to_Local(self, when):
 
     def __translate(self, condition, future=False, data=None):
         if future:
-            # Convert things like "sky is clear" to a future tense
-            try:
+            if (condition + ".future") in self.dialog_renderer.templates:
                 return self.dialog_renderer.render(condition + ".future", data)
-            except BaseException:
-                return condition
+        
+        if condition in self.dialog_renderer.templates:
+            return self.dialog_renderer.render(condition, data)
         else:
-            try:
-                return self.dialog_renderer.render(condition, data)
-            except BaseException:
-                return condition
-
+            return condition
+
+    """
+    OWM supports 31 languages, see https://openweathermap.org/current#multi
+    
+    If Mycroft's language setting is supported by OWM, 
+    then use it - or use 'en' otherwise
+    """
+    def __get_OWM_language(self, lang):
+        owmLang = 'en'
+        owmMulti = ['ar','bg','ca','cz','de','el','en','fa','fi','fr','gl',
+                    'hr','hu','it','ja','kr','la','lt','mk','nl','pl','pt',
+                    'ro','ru','se','sk','sl','es','tr','ua','vi']
+                    
+        lang = lang.lower()
+        LOG.debug("lang %s" % lang)
+        if lang == 'zh_zn' or lang == 'zh_tw':
+            owmLang = lang
+        elif lang[0:2] in owmMulti:
+            owmLang = lang[0:2]
+        LOG.debug("lang: %s owmLang: %s" % (lang, owmLang))
+        return owmLang
```

Comment:
Not sure this will work 100% accurately.

for swedish the language code in mycroft is `sv-se` which will not work with the lang[0:2].

Reply:
I went through the lang codes in OWM and most will work.

Maybe you could add special handling for the ones that doesn't work:
```
ISO	OWM	Language
sv-* 	se	Swedish
cs-cz	cz	Czech
ko-kr	kr	Korean
lv-lv	la	Latvian
uk-ua	ua	Ukrainan
```